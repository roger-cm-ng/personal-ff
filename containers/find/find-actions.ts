import { Dispatch } from 'redux';
import { postToken, postFind } from '../../helpers/resources';
import { FindBody } from './find-types';
import { ERROR_RAISED } from '../error-handler/error-handler-actions';
// eslint-disable-next-line import/no-cycle
import ErrorFindAgain from '../../components/error-content/error-find-again/error-find-again';

export const TOKEN_FETCHED = 'TOKEN_FETCHED';
export const FALCONE_FOUND = 'FALCONE_FOUND';

export const fetchToken = () => async (dispatch: Dispatch) => {
  try {
    const payload = await postToken();

    const { data, status } = payload;

    if (status >= 200 && status < 300) {
      dispatch({
        type: TOKEN_FETCHED,
        payload: data
      });
    } else {
      dispatch({
        type: ERROR_RAISED,
        component: ErrorFindAgain
      });
    }
  } catch (err) {
    dispatch({
      type: ERROR_RAISED,
      component: ErrorFindAgain
    });
  }
};

export const findFalcone = (body: FindBody) => async (dispatch: Dispatch) => {
  try {
    const payload = await postFind(body);

    const { data, status } = payload;

    if (status >= 200 && status < 300) {
      dispatch({
        type: FALCONE_FOUND,
        payload: data
      });
    } else {
      dispatch({
        type: ERROR_RAISED,
        component: ErrorFindAgain
      });
    }
  } catch (err) {
    dispatch({
      type: ERROR_RAISED,
      component: ErrorFindAgain
    });
  }
};
