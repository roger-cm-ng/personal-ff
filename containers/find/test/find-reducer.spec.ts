import findReducer from '../find-reducer';
import { FALCONE_FOUND } from './../find-actions';
import {
  FALCONE_FOUND_SUCCESS,
  FALCONE_FOUND_FAILURE
} from './find-mock-data';

describe('find reducer - success data', () => {
  const initialState = () => findReducer(null, {
    type: FALCONE_FOUND,
    payload: null
  });

  const acquiredState = () => findReducer(initialState(), {
    type: FALCONE_FOUND,
    payload: FALCONE_FOUND_SUCCESS
  });

  it('should return initial state', () => {
    expect(initialState()).toEqual(null);
  });

  it('should update the state with acquired data when type is FALCONE_FOUND', () => {
    expect(acquiredState()).toMatchSnapshot();
    expect(acquiredState()).toEqual(FALCONE_FOUND_SUCCESS);
  });
});

describe('find reducer - failure data', () => {
  const initialState = () => findReducer(null, {
    type: FALCONE_FOUND,
    payload: null
  });

  const acquiredState = () => findReducer(initialState(), {
    type: FALCONE_FOUND,
    payload: FALCONE_FOUND_FAILURE
  });

  it('should return initial state', () => {
    expect(initialState()).toEqual(null);
  });

  it('should update the state with acquired data when type is FALCONE_FOUND', () => {
    expect(acquiredState()).toMatchSnapshot();
    expect(acquiredState()).toEqual(FALCONE_FOUND_FAILURE);
  });
});