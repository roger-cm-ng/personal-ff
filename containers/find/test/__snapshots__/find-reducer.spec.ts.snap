// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`find reducer - failure data should update the state with acquired data when type is FALCONE_FOUND 1`] = `
Object {
  "status": "false",
}
`;

exports[`find reducer - success data should update the state with acquired data when type is FALCONE_FOUND 1`] = `
Object {
  "planet_name": "Jebing",
  "status": "success",
}
`;
